Express

arranger2222 opened the pull request:
Доброго дня, скидаю завдання по експресу та роутінгу. Відразу оговорюсь - є речі, які підглянув у інших студентів. Не знаю чому, але нода мені заходить втричі важче, ніж JS або Реакт. Тому була альтернатива -  скинути самому пізніше і підглянуте або ніяк. Дякую, врахую всі зауваження та побажання:)

arranger2222 committed:
init commit

arranger2222 committed:
finished task, fixied some bugs with routing

OnyshchenkoO commented on models/contacts.js:
```diff
@@ -1,19 +1,117 @@
+
+async function removeContact(contactId) {
+  try {
+    const contactsData = await fs.readFile(contactsPath, 'utf8');
+    const contactsList = JSON.parse(contactsData);
```
У тебе ці рядки повторюються у кожній функції. Винеси це як окрему функцію

OnyshchenkoO commented on models/contacts.js:
```diff
@@ -1,19 +1,117 @@
+
+    const [removedContact] = contactsList.splice(indexRemovedContact, 1);
+
+    const stringedContactsList = JSON.stringify(contactsList);
+
+    await fs.writeFile(contactsPath, stringedContactsList, 'utf8');
```
У тебе ці рядки повторюються у майже кожній функції. Винеси це як окрему функцію

arranger2222 committed:
Refactored code, fixed validation bugs

1 refactored write&read functions
2 refactored CRUD functions
3 changed nanoid instead uuid
4 fixed bugs with update schema
